tests/test_tar_extraction_security.py

```
@@ -0,0 +1,295 @@
+        extractall_calls: list[dict] = []
+
+        class SpyingTarFile:
+            def __init__(self, real_tar: tarfile.TarFile) -> None:
+                self._tar = real_tar
+
+            def getmembers(self):
+                return self._tar.getmembers()
+
+            def extractall(self, **kwargs):
+                extractall_calls.append(kwargs)
+                return self._tar.extractall(**kwargs)
```

Ruff (0.15.12)

[error] Uses of `tarfile.extractall()`

(S202)